File: figuras.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.println("Seleccione la figura geométrica a calcular:");
        System.out.println("1. Círculo");
        System.out.println("2. Rectángulo");
        System.out.println("3. Cuadrado");
        System.out.println("4. Triángulo");
        int opcion = scanner.nextInt();

        switch (opcion) {
            case 1:
                calcularCirculo(scanner);
                break;
            case 2:
                calcularRectangulo(scanner);
                break;
            case 3:
                calcularCuadrado(scanner);
                break;
            case 4:
                calcularTriangulo(scanner);
                break;
            default:
                System.out.println("Opción inválida.");
        }
        
        scanner.close();
    }
    
    private static void calcularCirculo(Scanner scanner) {
        System.out.println("Ingrese el radio del círculo:");
        double radio = scanner.nextDouble();
        System.out.println("Ingrese el color del círculo:");
        String color = scanner.next();
        
        Circulo circulo = new Circulo(radio, color);
        
        System.out.println("Área del círculo: " + circulo.getArea());
        System.out.println("Perímetro del círculo: " + circulo.getPerimetro());
        System.out.println("Color del círculo: " + circulo.getColor());
    }
    
    private static void calcularRectangulo(Scanner scanner) {
        System.out.println("Ingrese la base del rectángulo:");
        double base = scanner.nextDouble();
        System.out.println("Ingrese la altura del rectángulo:");
        double altura = scanner.nextDouble();
        System.out.println("Ingrese el color del rectángulo:");
        String color = scanner.next();
        
        Rectangulo rectangulo = new Rectangulo(base, altura, color);
        
        System.out.println("Área del rectángulo: " + rectangulo.getArea());
        System.out.println("Perímetro del rectángulo: " + rectangulo.getPerimetro());
        System.out.println("Color del rectángulo: " + rectangulo.getColor());
    }
    
    private static void calcularCuadrado(Scanner scanner) {
        System.out.println("Ingrese el lado del cuadrado:");
        double lado = scanner.nextDouble();
        System.out.println("Ingrese el color del cuadrado:");
        String color = scanner.next();
        
        Cuadrado cuadrado = new Cuadrado(lado, color);
        
        System.out.println("Área del cuadrado: " + cuadrado.getArea());
        System.out.println("Perímetro del cuadrado: " + cuadrado.getPerimetro());
        System.out.println("Color del cuadrado: " + cuadrado.getColor());
    }
    
    private static void calcularTriangulo(Scanner scanner) {
        System.out.println("Ingrese la base del triángulo:");
        double base = scanner.nextDouble();
        System.out.println("Ingrese la altura del triángulo:");
        double altura = scanner.nextDouble();
        System.out.println("Ingrese el tercer lado del triángulo:");
        double lado3 = scanner.nextDouble();
        System.out.println("Ingrese el color del triángulo:");
        String color = scanner.next();
        
        Triangulo triangulo = new Triangulo(base, altura, lado3, color);
        
        System.out.println("Área del triángulo: " + triangulo.getArea());
        System.out.println("Perímetro del triángulo: " + triangulo.getPerimetro());
        System.out.println("Color del triángulo: " + triangulo.getColor());
    }
}

class Circulo {
    private double radio;
    private String color;

    public Circulo(double radio, String color) {
        this.radio = radio;
        this.color = color;
    }

    public double getRadio() {
        return radio;
    }

    public String getColor() {
        return color;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double area() {
        return Math.PI * radio * radio;
    }

    private double perimetro() {
        return 2 * Math.PI * radio;
    }
}

class Rectangulo {
    private double base;
    private double altura;
    private String color;

    public Rectangulo(double base, double altura, String color) {
        this.base = base;
        this.altura = altura;
        this.color = color;
    }

    public double getBase() {
        return base;
    }

    public double getAltura() {
        return altura;
    }

    public String getColor() {
        return color;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double area() {
        return base * altura;
    }

    private double perimetro() {
        return 2 * (base + altura);
    }
}

class Cuadrado {
    private double lado;
    private String color;

    public Cuadrado(double lado, String color) {
        this.lado = lado;
        this.color = color;
    }

    public double getLado() {
        return lado;
    }

    public String getColor() {
        return color;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double area() {
        return lado * lado;
    }

    private double perimetro() {
        return 4 * lado;
    }
}

class Triangulo {
    private double base;
    private double altura;
    private double lado3;
    private String color;

    public Triangulo(double base, double altura, double lado3, String color) {
        this.base = base;
        this.altura = altura;
        this.lado3 = lado3;
        this.color = color;
    }

    public double getBase() {
        return base;
    }

    public double getAltura() {
        return altura;
    }

    public double getLado3() {
        return lado3;
    }

    public String getColor() {
        return color;
    }

    public double getArea() {
        return area();
    }

    public double getPerimetro() {
        return perimetro();
    }

    private double area() {
        return base * altura / 2;
    }

    private double perimetro() {
        return base + altura + lado3;
    }
}

File: figuras.py
import math

class HexagonoRegular:
    def __init__(self, lado, color):
        self.__lado = lado
        self.__color = color

    def get_lado(self):
        return self.__lado

    def get_color(self):
        return self.__color

    def get_area(self):
        return self.__area()

    def get_perimetro(self):
        return self.__perimetro()

    def __area(self):
        return (3 * math.sqrt(3) * self.__lado ** 2) / 2

    def __perimetro(self):
        return 6 * self.__lado

class Rombo:
    def __init__(self, diagonal_mayor, diagonal_menor, color):
        self.__diagonal_mayor = diagonal_mayor
        self.__diagonal_menor = diagonal_menor
        self.__color = color

    def get_diagonal_mayor(self):
        return self.__diagonal_mayor

    def get_diagonal_menor(self):
        return self.__diagonal_menor

    def get_color(self):
        return self.__color

    def get_area(self):
        return self.__area()

    def get_perimetro(self):
        return self.__perimetro()

    def __area(self):
        return (self.__diagonal_mayor * self.__diagonal_menor) / 2

    def __perimetro(self):
        return 4 * math.sqrt((self.__diagonal_mayor / 2) ** 2 + (self.__diagonal_menor / 2) ** 2)

class Trapecio:
    def __init__(self, base_mayor, base_menor, altura, color):
        self.__base_mayor = base_mayor
        self.__base_menor = base_menor
        self.__altura = altura
        self.__color = color

    def get_base_mayor(self):
        return self.__base_mayor

    def get_base_menor(self):
        return self.__base_menor

    def get_altura(self):
        return self.__altura

    def get_color(self):
        return self.__color

    def get_area(self):
        return self.__area()

    def get_perimetro(self):
        return self.__perimetro()

    def __area(self):
        return ((self.__base_mayor + self.__base_menor) * self.__altura) / 2

    def __perimetro(self):
        lado_oblicuo = math.sqrt(((self.__base_mayor - self.__base_menor) / 2) ** 2 + self.__altura ** 2)
        return self.__base_mayor + self.__base_menor + 2 * lado_oblicuo

def calcular_hexagono_regular():
    lado = float(input("Ingrese el lado del hexágono regular: "))
    color = input("Ingrese el color del hexágono regular: ")
    hexagono = HexagonoRegular(lado, color)
    print("Área del hexágono regular:", hexagono.get_area())
    print("Perímetro del hexágono regular:", hexagono.get_perimetro())
    print("Color del hexágono regular:", hexagono.get_color())

def calcular_rombo():
    diagonal_mayor = float(input("Ingrese la longitud de la diagonal mayor del rombo: "))
    diagonal_menor = float(input("Ingrese la longitud de la diagonal menor del rombo: "))
    color = input("Ingrese el color del rombo: ")
    rombo = Rombo(diagonal_mayor, diagonal_menor, color)
    print("Área del rombo:", rombo.get_area())
    print("Perímetro del rombo:", rombo.get_perimetro())
    print("Color del rombo:", rombo.get_color())

def calcular_trapecio():
    base_mayor = float(input("Ingrese la longitud de la base mayor del trapecio: "))
    base_menor = float(input("Ingrese la longitud de la base menor del trapecio: "))
    altura = float(input("Ingrese la altura del trapecio: "))
    color = input("Ingrese el color del trapecio: ")
    trapecio = Trapecio(base_mayor, base_menor, altura, color)
    print("Área del trapecio:", trapecio.get_area())
    print("Perímetro del trapecio:", trapecio.get_perimetro())
    print("Color del trapecio:", trapecio.get_color())

def main():
    print("Seleccione la figura geométrica a calcular:")
    print("1. Hexágono Regular")
    print("2. Rombo")
    print("3. Trapecio")
    opcion = int(input())

    if opcion == 1:
        calcular_hexagono_regular()
    elif opcion == 2:
        calcular_rombo()
    elif opcion == 3:
        calcular_trapecio()
    else:
        print("Opción inválida.")

if __name__ == "__main__":
    main()